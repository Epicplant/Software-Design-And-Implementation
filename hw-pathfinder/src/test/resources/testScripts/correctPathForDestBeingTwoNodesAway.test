# Tests that a dest node that is only two nodes away from the source node gives the correct answer

# Creates a new graph
CreateGraph newGraph

# Adds node A to the graph
AddNode newGraph A

# Adds node B to the graph
AddNode newGraph B

# Adds node C to the graph
AddNode newGraph C

# Adds node D to the graph
AddNode newGraph D

# Adds edge of weight 3 to the graph
AddEdge newGraph A B 3

# Adds edge of weight 3.888 to the graph
AddEdge newGraph A C 3.888

# Adds edge of weight 2.75 to the graph
AddEdge newGraph B C 2.75

# Adds edge of weight 12 to the graph
AddEdge newGraph C D 12

# Adds edge of weight 6 to the graph
AddEdge newGraph D C 6

# Adds edge of weight 4 to the graph
AddEdge newGraph C B 4

# Adds edge of weight 2 to the graph
AddEdge newGraph B A 2

# Finds the shortest path from node A to node D
FindPath newGraph A D

# Finds the shortest path from node D to node A
FindPath newGraph D A