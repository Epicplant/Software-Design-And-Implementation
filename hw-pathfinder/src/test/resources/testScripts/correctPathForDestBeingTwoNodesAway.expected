# Tests that a dest node that is only two nodes away from the source node gives the correct answer

# Creates a new graph
created graph newGraph

# Adds node A to the graph
added node A to newGraph

# Adds node B to the graph
added node B to newGraph

# Adds node C to the graph
added node C to newGraph

# Adds node D to the graph
added node D to newGraph

# Adds edge of weight 3 to the graph
added edge 3.000 from A to B in newGraph

# Adds edge of weight 3.888 to the graph
added edge 3.888 from A to C in newGraph

# Adds edge of weight 2.75 to the graph
added edge 2.750 from B to C in newGraph

# Adds edge of weight 12 to the graph
added edge 12.000 from C to D in newGraph

# Adds edge of weight 6 to the graph
added edge 6.000 from D to C in newGraph

# Adds edge of weight 4 to the graph
added edge 4.000 from C to B in newGraph

# Adds edge of weight 2 to the graph
added edge 2.000 from B to A in newGraph

# Finds the shortest path from node A to node D
path from A to D:
A to C with weight 3.888
C to D with weight 12.000
total cost: 15.888

# Finds the shortest path from node D to node A
path from D to A:
D to C with weight 6.000
C to B with weight 4.000
B to A with weight 2.000
total cost: 12.000